pkg/apiserver/api_installer.go

```diff
@@ -323,7 +323,11 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		actions = appendIf(actions, action{"PATCH", itemPath, nameParams, namer}, isPatcher)
 		actions = appendIf(actions, action{"DELETE", itemPath, nameParams, namer}, isDeleter)
 		actions = appendIf(actions, action{"WATCH", "watch/" + itemPath, nameParams, namer}, isWatcher)
+		// We add "proxy" subresource to remove the need for the generic top level prefix proxy.
+		// The generic top level prefix proxy will be deprecated for v1.2.
```

Comment:
"The generic top level prefix proxy is deprecated in v1.2, and will be removed in 1.3, or 1.4 at the latest."